actors/miner/src/lib.rs

```diff
@@ -2695,6 +2692,196 @@ impl Actor {
         Ok(())
     }
 
+    /// FYI: https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0070.md
+    fn move_partitions(rt: &impl Runtime, params: MovePartitionsParams) -> Result<(), ActorError> {
+        if params.orig_deadline == params.dest_deadline {
+            return Err(actor_error!(illegal_argument, "orig_deadline == dest_deadline"));
+        }
+        let policy = rt.policy();
+        if params.orig_deadline >= policy.wpost_period_deadlines
+            || params.dest_deadline >= policy.wpost_period_deadlines
+        {
+            return Err(actor_error!(
+                illegal_argument,
+                "invalid param, orig_deadline: {}, dest_deadline: {}",
+                params.orig_deadline,
+                params.dest_deadline
+            ));
+        }
+        if params.partitions.is_empty() {
+            return Err(actor_error!(illegal_argument, "empty partitions not allowed"));
+        }
+
+        rt.transaction(|state: &mut State, rt| {
+            let info = get_miner_info(rt.store(), state)?;
+
+            rt.validate_immediate_caller_is(
+                info.control_addresses.iter().chain(&[info.worker, info.owner]),
+            )?;
+
+            let store = rt.store();
+            let current_deadline = state.deadline_info(policy, rt.curr_epoch());
+            let mut deadlines = state
+                .load_deadlines(store)
+                .context_code(ExitCode::USR_ILLEGAL_STATE, "failed to load deadlines")?;
+
+            ensure_deadline_available_for_move(
+                policy,
+                params.orig_deadline,
+                params.dest_deadline,
+                &current_deadline,
+            )
+            .context_code(
+                ExitCode::USR_FORBIDDEN,
+                "conditions not satisfied for deadline_available_for_move",
+            )?;
+
+            let mut orig_deadline =
+                deadlines.load_deadline(store, params.orig_deadline).context_code(
+                    ExitCode::USR_ILLEGAL_STATE,
+                    format!("failed to load deadline {}", params.orig_deadline),
```

Comment:
Nit: use `with_context_code` when doing a `format!` operation for the message

Reply:
Done, I guess this is to avoid alloc when no error happens?

Reply:
Yes.